### PR TITLE
chore(docker): add redpanda for docker compose

### PR DESCRIPTION
**Pull request by Gogomoe**
## What's changed and what's your intention?

Fix command of start compute node.
Add redpanda service.

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests

## Refer to a related PR or issue link (optional)

#1606

### CODE REVIEW

File: docker/README.md [hunk 1]
```diff
@@ -10,5 +10,13 @@ make docker
 
 in the project root.
 
+To start entire app, run
+
+```
+make docker-compose
```
Comment (neverchanje) [inline docker/README.md:16]: Are we default to redpanda? In the later development, we may need to test multiple source systems. So maybe we should change the name to:

```
make docker-compose-redpanda
```
`docker-compose-kafka`, `docker-compose-pulsar`, etc.

File: docker/docker-compose.yml [hunk 2]
```diff
@@ -25,19 +25,26 @@ services:
...
     depends_on:
       - metanode
-      - computenode
\ No newline at end of file
+      - computenode
+  redpanda:
+    container_name: redpanda_container
+    network_mode: host
+    image: docker.vectorized.io/vectorized/redpanda:latest
+    command: redpanda start --smp 1 --reserve-memory 0M --overprovisioned --node-id 0
```
Comment (tabVersion) [inline docker/docker-compose.yml:50]: should we specify `kafka-addr` and `advertise-kafka-addr` here as mentioned in [their docker quickstart](https://docs.redpanda.com/docs/quickstart/quick-start-docker/#bring-up-a-docker-compose-file)

File: docker/docker-compose.yml [hunk 3]
```diff
@@ -25,19 +25,26 @@ services:
...
     image: ghcr.io/singularity-data/risingwave:latest
-    command: /bin/bash -c /risingwave/bin/frontend-v2 --host 127.0.0.1:4566 --meta-addr http://127.0.0.1:5690
+    command: /risingwave/bin/frontend-v2 --host 127.0.0.1:4566 --meta-addr http://127.0.0.1:5690
     depends_on:
       - metanode
-      - computenode
\ No newline at end of file
+      - computenode
+  redpanda:
+    container_name: redpanda_container
```
Comment (neverchanje) [inline docker/docker-compose.yml:47]: Im considering making the yml as a base template and adding different source systems separately via their config file. 

```
cat redpanda.yml > docker-compose.yml
```

But bascially this pr LGTM.

**Review by neverchanje — APPROVED** [review]

File: docker/docker-compose.yml (hunk 2 above)
Comment (Gogomoe) [reply docker/docker-compose.yml]: agree. committed.

File: docker/README.md (hunk 1 above)
Comment (Gogomoe) [reply docker/README.md:16]: I think we can do it when add other source.

**Review by twocode — APPROVED** [review]
I also think `make docker-compoase-redpanda` would be more meaningful, considering we'll have ecosystem partnered with it. 
LGTM. Great work!

**Comment by pangzhenzhou** [comment]
LGTM. Good job. Consider passing configuration users may modify to the container via environment variables, such as ports.  But this should be optimization can be done in another PR.

https://docs.docker.com/compose/environment-variables/

**Pull request merged**